src/utils/OnboardingProvider.tsx

```diff
@@ -76,11 +79,12 @@ export default function OnboardingProvider({
     };
 
     const flushData = async () => {
+      if (!auth) throw new Error('Fatal: No auth context provided!');
       if (!profileCtx) throw new Error('Fatal: No profile context provided!');
-      if (profileCtx.userId === undefined)
+      if (auth.userId === undefined)
         throw new Error('Fatal: User is not logged in!');
 
-      const uid: UUID = profileCtx.userId;
+      const uid: UUID = auth.userId;
 
```

Comment:
can you add `auth` to either:
- the dependency array of the `useMemo` hook; or
- if you decide to implement the `useCallback` hooks here, then the dependency array of that

Reply:
added it to the dependency array of the useMemo hook!